pkg/container/writer.go

```diff
@@ -25,84 +24,100 @@ func (ctn Writer) AddSealed(cid cid.Cid, data []byte) {
 	ctn[cid] = data
 }
 
-const currentContainerVersion = "ctn-v1"
+// ToBytes encode the container into raw bytes.
+func (ctn Writer) ToBytes() ([]byte, error) {
+	return ctn.toBytes(headerRawBytes)
+}
 
-// ToCbor encode the container into a DAG-CBOR binary format.
-func (ctn Writer) ToCbor() ([]byte, error) {
-	var buf bytes.Buffer
-	err := ctn.ToCborWriter(&buf)
-	if err != nil {
-		return nil, err
-	}
-	return buf.Bytes(), nil
+// ToBytesWriter is the same as ToBytes, but with an io.Writer.
+func (ctn Writer) ToBytesWriter(w io.Writer) error {
+	return ctn.toWriter(headerRawBytes, w)
 }
 
-// ToCborWriter is the same as ToCbor, but with an io.Writer.
-func (ctn Writer) ToCborWriter(w io.Writer) error {
-	node, err := qp.BuildMap(basicnode.Prototype.Any, 1, func(ma datamodel.MapAssembler) {
-		qp.MapEntry(ma, currentContainerVersion, qp.List(int64(len(ctn)), func(la datamodel.ListAssembler) {
-			for _, data := range ctn {
-				qp.ListEntry(la, qp.Bytes(data))
-			}
-		}))
-	})
-	if err != nil {
-		return err
-	}
-	return ipld.EncodeStreaming(w, node, dagcbor.Encode)
+// ToBytesGzipped encode the container into gzipped bytes.
+func (ctn Writer) ToBytesGzipped() ([]byte, error) {
```

Comment:
The number of methods created could explode with enough permutations of encoding and compression.  Can we instead add options to `ToBytes` and `ToWriter` that set the encoding and compression?  Or perhaps pass in the sigil as a second argument?  Which encoding/compression is the default or convention?  Will most people create raw (unencoded/uncompressed) `container`s simply because `ToBytes` seems simplest?

Reply:
I thought about doing that, but really the number of combination should remain low. The way you suggest would mean exposing the `header` type, which feels too much. The rest of the codebase has this style, not much reason to go against it.